src/os/bluestore/BitAllocator.cc

```diff
@@ -989,7 +983,7 @@ inline bool BitMapAreaLeaf::child_check_n_lock(BitMapZone* const child,
   /* The exhausted check can be performed without acquiring the lock. This
    * is because 1) BitMapZone::is_exhausted() actually operates atomically
    * and 2) it's followed by the exclusive, required-aware re-verification. */
-  if (child->is_exhausted()) {
+  if (child->BitMapZone::is_exhausted()) {
```

Comment:
As the method is declared now as `final` in `BitAllocator.h`, compiler already gets a chance to skip the `vtable` check. Enforcing this behaviour via `->BitMapZone::` is just for those ones that don't apply this optimization (if any).